workflows/illumina.nf

```diff
@@ -107,16 +166,102 @@ workflow ILLUMINA {
   POST_TABLE2ASN(TABLE2ASN.out.genbank)
   ch_versions = ch_versions.mix(POST_TABLE2ASN.out.versions)
 
-  BLAST_BLASTN(IRMA.out.consensus, BLAST_MAKEBLASTDB.out.db)
-  ch_versions = ch_versions.mix(BLAST_BLASTN.out.versions.first().ifEmpty(null))
+  BLAST_BLASTN_IRMA(IRMA.out.consensus, BLAST_MAKEBLASTDB_NCBI.out.db)
+  ch_versions = ch_versions.mix(BLAST_BLASTN_IRMA.out.versions.first().ifEmpty(null))
 
-  ch_blast = BLAST_BLASTN.out.txt.collect({ it[1] })
-  SUBTYPING_REPORT(
+  ch_blast = BLAST_BLASTN_IRMA.out.txt.collect({ it[1] })
+  SUBTYPING_REPORT_IRMA_CONSENSUS(
     ZSTD_DECOMPRESS_CSV.out.file,
     ch_blast,
     CHECK_SAMPLE_SHEET.out
   )
-  ch_versions = ch_versions.mix(SUBTYPING_REPORT.out.versions)
+  ch_versions = ch_versions.mix(SUBTYPING_REPORT_IRMA_CONSENSUS.out.versions)
+
+  // Prepare top ncbi accession id for each segment of each sample (id which has top bitscore)
+  PULL_TOP_REF_ID(BLAST_BLASTN_IRMA.out.txt, ZSTD_DECOMPRESS_CSV.out.file)
+  ch_versions = ch_versions.mix(PULL_TOP_REF_ID.out.versions)
+
+  PULL_TOP_REF_ID.out.accession_id
+    .map { it[1] }
+    .splitCsv(header: false, sep:",")
+    .map{ [it[0], it[1], it[2]] }
+    .combine(CAT_ILLUMINA_FASTQ.out.reads.map { [it[0].id, it[1]] }, by: 0)
+    .set { ch_sample_segment } 
+
+  // Pull segment reference sequence for each sample
+  SEQTK_SEQ(ch_sample_segment, ZSTD_DECOMPRESS_FASTA.out.file)
+  ch_versions = ch_versions.mix(SEQTK_SEQ.out.versions)
+
+  // Map reads against segment reference sequences using Minimap2
+  MINIMAP2(SEQTK_SEQ.out.sample_info)
+  ch_versions = ch_versions.mix(MINIMAP2.out.versions)
+
+  // Generate coverage and stats
+  MOSDEPTH_GENOME(MINIMAP2.out.alignment)
+  ch_versions = ch_versions.mix(MOSDEPTH_GENOME.out.versions)
+
+  FREEBAYES(MINIMAP2.out.alignment)
+  ch_versions = ch_versions.mix(FREEBAYES.out.versions)
+
+  BCF_FILTER_FREEBAYES(FREEBAYES.out.vcf, params.major_allele_fraction)
+  ch_versions = ch_versions.mix(BCF_FILTER_FREEBAYES.out.versions)
+  ch_vcf_filter = BCF_FILTER_FREEBAYES.out.vcf
+
+  VCF_FILTER_FRAMESHIFT(ch_vcf_filter)
```

Comment:
This might only be applicable for Nanopore reads which aren't great in homopolymer regions.

Reply:
@peterk87 After some discussion with Cody, I think it's best to keep this function as it is as it can't hurt. While frameshift errors are more commonly associated with Nanopore reads, this filter can still be applicable, albeit rarely, for Illumina data as well. The function's purpose of identifying INDELs not divisible by three is useful for ensuring the integrity of the data, even if the occurrence is infrequent for Illumina reads.